frontend/src/pages/modelServing/screens/projects/kServeModal/ManageKServeModal.tsx

```diff
@@ -185,6 +186,29 @@ const ManageKServeModal: React.FC<ManageKServeModalProps> = ({
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [createDataInferenceService.name, setKserveNameDesc]);
 
+  React.useEffect(() => {
+    if (modelDeployPrefillInfo) {
+      const isDataReady =
+        !!kServeNameDesc.name &&
+        Array.isArray(servingRuntimeTemplates) &&
+        servingRuntimeTemplates.length > 0 &&
+        (!projectContext || connectionsLoaded) &&
+        (podSpecOptionsState.acceleratorProfile.loaded ||
+          podSpecOptionsState.hardwareProfile.profilesLoaded);
+
+      setIsLoading(!isDataReady);
+    }
+  }, [
+    kServeNameDesc.name,
+    servingRuntimeTemplates,
+    projectTemplates,
+    projectContext,
+    connectionsLoaded,
+    modelDeployPrefillInfo,
+    podSpecOptionsState.acceleratorProfile.loaded,
+    podSpecOptionsState.hardwareProfile.profilesLoaded,
+  ]);
+
```

Comment:
`useEffect` scares me now after we have fixed multiple issues with prefilling info, can we wrap everything into a `useMemo` and have a `useEffect` on it? Something like:
```suggestion
  const isDataReady = React.useMemo(() => {
    if (!modelDeployPrefillInfo) {
      return false;
    }

    return (
      !!kServeNameDesc.name &&
      Array.isArray(servingRuntimeTemplates) &&
      servingRuntimeTemplates.length > 0 &&
      (!projectContext || connectionsLoaded) &&
      (podSpecOptionsState.acceleratorProfile.loaded ||
        podSpecOptionsState.hardwareProfile.profilesLoaded)
    );
  }, [
    kServeNameDesc.name,
    servingRuntimeTemplates,
    projectContext,
    connectionsLoaded,
    modelDeployPrefillInfo,
    podSpecOptionsState.acceleratorProfile.loaded,
    podSpecOptionsState.hardwareProfile.profilesLoaded,
  ]);

  React.useEffect(() => {
    setIsLoading(!isDataReady);
  }, [isDataReady]);
```